src/Symfony/Component/Asset/Package.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset;
+
+use Symfony\Component\Asset\Context\ContextInterface;
+use Symfony\Component\Asset\VersionStrategy\VersionStrategyInterface;
+
+/**
+ * Basic package that adds a version to asset URLs.
+ *
+ * @author Kris Wallsmith <kris@symfony.com>
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class Package implements PackageInterface
+{
+    private $versionStrategy;
+    private $context;
+
+    public function __construct(VersionStrategyInterface $versionStrategy)
+    {
+        $this->versionStrategy = $versionStrategy;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getVersion($path)
+    {
+        return $this->versionStrategy ? $this->versionStrategy->getVersion($path) : '';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getUrl($path)
+    {
+        if ($this->isAbsoluteUrl($path)) {
+            return $path;
+        }
+
+        return $this->versionStrategy->applyVersion($path);
+    }
+
+    public function setContext(ContextInterface $context)
+    {
+        $this->context = $context;
+    }
+
+    protected function getContext()
```

Comment:
you should document it as ``@return ContextInterface|null``. It is important for child classes